### DISCUSSION TITLE
`next dev`: NOT Ignoring TypeScript Errors

### DISCUSSION THREAD

**Discussion by mktcode** [vercel/next.js#51843] [Help] [answered]
I want the opposite of this:
https://nextjs.org/docs/app/building-your-application/configuring/typescript#ignoring-typescript-errors

`next dev` should fail when there are ts errors that would occur when `next build`ing.

I want to ensure that `next build` will always work, by not allowing to commit ts errors. A pre-commit hook feels potentially a bit annoying and "too late".

**Comment by icyJoseph** [comment, marked as answer]
I think this thread explains a whole lot on why this ended up not being that way. 

https://github.com/vercel/next.js/issues/14997

And it also has some work arounds.

- This one seems to be pretty good: https://github.com/vercel/next.js/issues/14997#issuecomment-726692314 (might benefit from the npm concurrently package)
- https://github.com/vercel/next.js/issues/14997#issuecomment-772649119
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack(config, { isServer }) {
    if (!isServer) {
      const ForkTsCheckerWebpackPlugin = require("fork-ts-checker-webpack-plugin");
      config.plugins.push(new ForkTsCheckerWebpackPlugin());
    }
    return config;
  },
};

module.exports = nextConfig;

``` 

And install, https://github.com/TypeStrong/fork-ts-checker-webpack-plugin:

```console
yarn add --dev fork-ts-checker-webpack-plugin
```

And then running `next dev` will also catch the errors in dev time. 

<img width="1016" alt="Screenshot 2023-06-26 at 22 45 50" src="https://github.com/vercel/next.js/assets/21013447/443f09ad-745d-4081-b147-9fed9dfc7579">

**Reply by mktcode** [reply]
Thanks a lot!

**Reply by codler** [reply]
How do nextjs typecheck on build without fork-ts-checker-webpack-plugin?
